### PR TITLE
Fix text-only batches for multimodal training

### PR DESCRIPTION
**Pull request by ybochkov**
Treat image inputs as optional when `use_multimodal` is enabled so text-only batches can train while retaining vision parameters in checkpoints. Add a regression test for the NNX training path.

# Description

This PR allows a multimodal model to process text-only training batches that do not contain `images` or `image_masks`.

Previously, `loss_fn` accessed `data["images"]` whenever `use_multimodal` was enabled. This raised a `KeyError` for text-only batches even though the model forward paths already support `encoder_images=None`.

This is needed when fine-tuning the language portion of a multimodal model on text-only data while retaining the pretrained vision parameters in the resulting checkpoint. Enabling `use_multimodal` constructs the multimodal model and includes its vision parameters in initialization and checkpointing; it does not require every training batch to contain image data.

The implementation:

- Retrieves `images` and `image_masks` using optional dictionary lookups.
- Passes `None` when either input is absent.
- Applies the same behavior to the Flax Linen and NNX paths.
- Documents why a multimodal model may receive a text-only batch.
- Adds an NNX regression test that enables multimodal support and calls `loss_fn` with no image fields.

Batches containing images continue to behave as before, and models with `use_multimodal=false` are unaffected.

This change does not modify vision-encoder freezing, optimizer behavior, or checkpoint structure. It only makes image inputs optional at the training forward-pass boundary.

# Tests

The complete NNX training unit-test module passes locally on CPU:

```bash
source .venv/bin/activate
JAX_PLATFORMS=cpu python -m pytest -v tests/unit/train_nnx_test.py

Result: 14 passed.

The new regression test also passes with branch coverage enabled:

JAX_PLATFORMS=cpu python -m pytest -v \
  tests/unit/train_nnx_test.py::TestLossFnNNX::test_multimodal_model_accepts_text_only_batch \
  --cov=maxtext \
  --cov-branch \
  --cov-report=term-missing
```

Result: 1 passed.

End-to-end testing was not run because this is a narrowly scoped input-handling change covered by the focused unit test.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):

- [x] I have performed a self-review of my code. For an optional AI review, add the gemini-review label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as
  explained in our documentation (https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files). No documentation changes are needed because this restores behavior already supported by the model interface and does not introduce new configuration or user-facing APIs.

### CODE REVIEW

**Review by shralex — APPROVED** [review]

**Review by aireenmei — APPROVED** [review]

**Review by aireenmei — COMMENTED** [review]
Please complete the CLA according to the link mentioned in cla bot

**Comment by ybochkov** [comment]
@aireenmei, thank you for your assistance!
CLA has been signed and PR has been rescanned. Looks like `cla/google` check is passed now. Anything else is still required on my end?

**Comment by ybochkov** [comment]
@shralex, @aireenmei, as far as I can see, there are two pending checks awaiting approval from a maintainer. I was wondering if there’s anything blocking us from launching them. Thanks!
